pkg/bootstrap/docker/up.go

```diff
@@ -303,6 +305,22 @@ func (c *ClientStartConfig) CreateDockerMachine(out io.Writer) error {
 	return dockermachine.NewBuilder().Name(c.DockerMachine).Create()
 }
 
+// CheckOpenShiftClient ensures that the client can be configured
+// for the new server
+func (c *ClientStartConfig) CheckOpenShiftClient(out io.Writer) error {
+	kubeConfig := os.Getenv("KUBECONFIG")
+	if len(kubeConfig) == 0 {
+		return nil
+	}
+	file, err := os.OpenFile(kubeConfig, os.O_RDWR, 0644)
+	if err != nil {
+		return errors.ErrKubeConfigNotWriteable(kubeConfig, err)
+	}
+	file.Close()
```

Comment:
Potentially interesting possible golang issue.  At least with 1.6.

Here is the end of `os.OpenFile`:
```
	return NewFile(uintptr(r), name), nil
```
Here is the impl of `os.NewFile`:

```
func NewFile(fd uintptr, name string) *File {
	fdi := int(fd)
	if fdi < 0 {
		return nil
	}
	f := &File{&file{fd: fdi, name: name}}
	runtime.SetFinalizer(f.file, (*file).close)
	return f
}
```

It is too remote / too much of a nit to change @csrwng 's  code here
but technically speaking, if the file descriptor number is large enough, you could end up with `OpenFile` returning nil, nil